apis/src/responses/challenge.rs

```diff
@@ -70,5 +72,77 @@ impl ChallengeResponse {
             band_lower: challenge.band_lower,
         })
     }
+
+}
+}
+}
+
+fn is_compatible(challenge: &ChallengeResponse, user: &str, details: &ChallengeDetails) -> bool {
+    details.game_type == GameType::from_str(&challenge.game_type).unwrap()
+        && details.rated == challenge.rated
+        && details.time_mode == challenge.time_mode
+        && details.time_base == challenge.time_base
+        && details.time_increment == challenge.time_increment
+        && match details.color_choice {
+            ColorChoice::Random => true,
+            ColorChoice::White => {
+                challenge.color_choice == ColorChoice::Black
+                    || challenge.color_choice == ColorChoice::Random
+            }
+            ColorChoice::Black => {
+                challenge.color_choice == ColorChoice::White
+                    || challenge.color_choice == ColorChoice::Random
+            }
+        }
+        && if let Some(opponent) = &details.opponent {
+            opponent == &challenge.challenger.username
+        } else if let Some(opponent) = &challenge.opponent {
+            opponent.username == user
+        } else {
+            challenge.challenger.username != user
+        }
+}
+
+fn has_same_details(challenge: &ChallengeResponse, user: &str, details: &ChallengeDetails) -> bool {
```

Comment:
Something like this might be good thing to add but only for realtime games. Creating two games with identical settings for correspondence shouldn't be stopped. Maybe I want to start playing correspondence more seriously I should be allowed to open many challenges at once.

Reply:
Good point, unlimited time games and correspondence should not have this restriction